cooperator_portal/controllers/portal.py

```diff
@@ -15,20 +15,14 @@
 
 class CooperatorPortalAccount(CustomerPortal):
     def __init__(self, *args, **kwargs):
-        super().__init__()
-        # Class scope is accessible throughout the server even on
-        # odoo instances that do not install this module.
-
-        # Therefore : bring back to instance scope if not already
-        #  if "MANDATORY_BILLING_FIELDS" in vars(self):
-        if "MANDATORY_BILLING_FIELDS" not in vars(self):
-            self.MANDATORY_BILLING_FIELDS = (
-                CustomerPortal.MANDATORY_BILLING_FIELDS.copy()
-            )
-
-        self.MANDATORY_BILLING_FIELDS.extend(
-            ["iban", "birthdate_date", "gender", "lang"]
-        )
+        super().__init__(*args, **kwargs)
+        # these are class constants (shared between all running odoo
+        # instances), so they have to be copied to self to be modified only
+        # for the odoo instances using this module.
+        self.MANDATORY_BILLING_FIELDS = self.MANDATORY_BILLING_FIELDS.copy()
```

Comment:
Why getting rid of the `if "MANDATORY_BILLING_FIELDS" not in vars(self):` ?

Reply:
because this is a useless test. ☺️ python guarantees that they are there, because these class variables are defined in the class that this class extends.

Reply:
i changed my mind: i left the test, in case a sub-class already copied the variable to self.